docs/fsharp/style-guide/formatting.md

```diff
@@ -808,7 +808,89 @@ let pascalsTriangle =
        [| 1; 8; 28; 56; 70; 56; 28; 8; 1 |] |]
 ```
 
-And as with records, declaring the opening and closing brackets on their own line will make moving code around and piping into functions easier.
+As with records, declaring the opening and closing brackets on their own line will make moving code around and piping into functions easier:
+
+```fsharp
+// ✔️ OK
+let pascalsTriangle =
+    [| 
+        [| 1 |]
+        [| 1; 1 |]
+        [| 1; 2; 1 |]
+        [| 1; 3; 3; 1 |]
+        [| 1; 4; 6; 4; 1 |]
+        [| 1; 5; 10; 10; 5; 1 |]
+        [| 1; 6; 15; 20; 15; 6; 1 |]
+        [| 1; 7; 21; 35; 35; 21; 7; 1 |]
+        [| 1; 8; 28; 56; 70; 56; 28; 8; 1 |] 
+    |]
+```
+
+If a list or array expression is the right-hand side of a binding, you may prefer to use "Stroustrup" style:
+
+```fsharp
+// ✔️ OK
+let pascalsTriangle = [| 
+   [| 1 |]
+   [| 1; 1 |]
+   [| 1; 2; 1 |]
+   [| 1; 3; 3; 1 |]
+   [| 1; 4; 6; 4; 1 |]
+   [| 1; 5; 10; 10; 5; 1 |]
+   [| 1; 6; 15; 20; 15; 6; 1 |]
+   [| 1; 7; 21; 35; 35; 21; 7; 1 |]
+   [| 1; 8; 28; 56; 70; 56; 28; 8; 1 |] 
+|]
+```
+
+However, when a list or array expression is _not_ the right-hand side of a binding, such as when it's inside of another list or array, if that inner expression needs to span multiple lines, the brackets should go on their own lines:
+
+```fsharp
+// ✔️ OK - The outer list follows "Stroustrup" style, while the inner lists place their brackets on separate lines
+let fn a b = [ 
+    [
+        someReallyLongValueThatWouldForceThisListToSpanMultipleLines
+        a
+    ]
+    [ 
+        b
+        someReallyLongValueThatWouldForceThisListToSpanMultipleLines 
+    ]
+]
+
+// ❌ Not okay
+let fn a b = [ [
+    someReallyLongValueThatWouldForceThisListToSpanMultipleLines
+    a
+]; [
+    b
+    someReallyLongValueThatWouldForceThisListToSpanMultipleLines
+] ]
+```
```

Comment:
Semi-related is elmish.  How should we go about `function list1 list2`? I've seen it typically like:

```fsharp
button [
    ClassName ("btn " + if model.NewBook.Validate() then "btn-primary" else "btn-disabled")
    OnClick (fun _ -> dispatch ValidateBook) ] [
    i [ClassName "glyphicon glyphicon-plus"; Style [PaddingRight 5]] []
    str "Add"
]
```

where the `] [` is on the same line.

Reply:
I think in function applications you can justify the `] [` placement as it does not depend on the `;` being present.

Reply:
I'm not super familiar with the intricacies of elmish code, having never used it. I'd welcome suggestions on how to address that style here

Reply:
We could address this in a future PR, though related, it is a bit of a subject of its own.
I think the rules which Fantomas applies today to format this are a good start.